tests/integration/mock_server.py

```diff
@@ -0,0 +1,54 @@
+import time
+import uuid
+import uvicorn
+
+from contextlib import contextmanager
+from fastapi import FastAPI, Request, Response
+from multiprocessing import Process
+from typing import Optional
+
+from redact import JobArguments, JobPostResponse
+
+
+@contextmanager
+def mock_redact_server(expected_path: Optional[str] = None, expected_job_args: Optional[JobArguments] = None):
+    """
+    Context manager that starts a mock Redact server (127.0.0.1:8787) which returns a 500 error when the request does
+    not look as expected.
+    """
+
+    app = FastAPI()
+
+    @app.post('/{path:path}')
+    def get_all_routes(request: Request):
+        return _mock_redact_request_handler(request=request,
+                                            expected_path=expected_path,
+                                            expected_job_args=expected_job_args)
+
+    server = Process(target=uvicorn.run, args=(app,), kwargs={'port': 8787}, daemon=True)
+    server.start()
+    time.sleep(2)  # wait for server to be started
+    yield
+    server.terminate()
+    server.join()
```

Comment:
what is this join doing after the terminate?

Reply:
Waiting for the service to shut down. `terminate()` only sends a `SIGTERM`.